### PR TITLE
Fix end-user-facing uses of inline CSS

### PR DESCRIPTION
**Pull request by ClearlyClaire**
Mastodon currently relies on inline CSS rules quite a bit, which is often suboptimal code factorization, and overall prevents us from using stricter Content-Security Policies.

~~This PR removes *most* of the inline CSS attributes an end-user would run into with the following exceptions:~~
- ~~poll vote percentage bars in the JS-less fallback, as there doesn't seem to be any decent way to do this~~
- ~~the account migration banner thing, as that uses background images~~
- ~~some styling on the two-factor auth screen as I'm not sure where to put that in~~

It also doesn't fix most of the issues specific to the admin interface.

EDIT: used the `progress` HTML tag for fallback for poll options, successfully tested with recent Chromium and Firefox versions, as well as with an antiquated Firefox version
EDIT#2: changed the “moved” notice to use image tags
EDIT#3: changed the 2FA styling as well

### CODE REVIEW

**Review by ykzts — APPROVED** [review]

File: app/views/settings/preferences/appearance/show.html.haml [hunk 1]
```diff
@@ -12,8 +12,8 @@
       = f.input :setting_theme, collection: Themes.instance.names, label_method: lambda { |theme| I18n.t("themes.#{theme}", default: theme) }, wrapper: :with_label, include_blank: false, hint: false
 
   - unless I18n.locale == :en
-    .flash-message{ style: "text-align: unset; color: unset" }
-      #{t 'appearance.localization.body'} #{content_tag(:a, t('appearance.localization.guide_link_text'), href: t('appearance.localization.guide_link'), target: "_blank", rel: "noopener", style: "text-decoration: underline")}
+    .flash-message
```
Comment (Sasha-Sorokin) [inline app/views/settings/preferences/appearance/show.html.haml:15]: `.translation-prompt` class seems missing? :thinking: :eyes:

Comment (ThibG) [reply app/views/settings/preferences/appearance/show.html.haml]: Indeed, thanks!

**Comment by ThibG** [comment]
No end-user-facing inline CSS should be left, which means that with this PR, `unsafe-inline` can be removed and not affect users.

However, the admin interface still makes use of inline CSS quite a bit.

File: app/views/statuses/_poll.html.haml [hunk 2]
```diff
@@ -16,7 +16,8 @@
             %span.poll__option__text
               = Formatter.instance.format_poll_option(status, option, autoplay: autoplay)
 
-          %span.poll__chart{ style: "width: #{percent}%" }
+          %progress{ max: 100, value: percent < 1 ? 1 : percent }
```
Comment (Sasha-Sorokin) [inline app/views/statuses/_poll.html.haml:20]: This makes bar accessible by screen reader software, which could have been a good thing if only it wouldn't repeat [the same thing above](https://github.com/tootsuite/mastodon/blob/2c9038b416713bf2793803d16643c37df08e30da/app/views/statuses/_poll.html.haml#L15) and confuse user with its role (*progress* bar). Should it be `aria-hidden` from a11y-relying users?

**Review by Gargron — APPROVED** [review]

**Pull request merged**